app/viewmodels/components/InputCharacterCountViewModel.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.components
+
+import play.twirl.api.Html
+
+sealed trait InputCharacterCountViewModel
+
+object InputCharacterCountViewModel {
+
+  case class OrdinaryInputCharacterCount(
+    heading: String,
+    caption: Option[String] = None
+  ) extends InputCharacterCountViewModel
+
+  case class InputCharacterCountWithHiddenLabel(
```

Comment:
Would `...WithAdditionalHtml` be more consistent with our other view models than `...WithHiddenLabel`?